### PR TITLE
Modular Views

### PR DESCRIPTION
**Pull request by ConnorVG**
The aim of this PR is the achieve a modular view system. I don't believe this should be part of the plugin PR.

### CODE REVIEW

File: app/Foundation/Providers/ModuleServiceProvider.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    /**
+     * Boot the service provider.
+     *
+     * @param \Illuminate\View\Compilers\BladeCompiler $blade
+     */
+    public function boot(BladeCompiler $blade)
+    {
+        $blade->directive('modules', function ($group = null) {
+            $code = [
+                'foreach (array_numeric_sort($modules) as $group => $parts) {',
```
Comment (ConnorVG) [inline app/Foundation/Providers/ModuleServiceProvider.php]: I want this tidied and turned into a single recursive function call of some sort.

File: app/Foundation/Providers/ModuleServiceProvider.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+     * @var array
+     */
+    protected $groups = [
+        'index' => [
+            'messages'   => 10000,
+            'status'     => 20000,
+            'components' => 30000,
+            'metrics'    => 40000,
+            'scheduled'  => 50000,
+            'timeline'   => 60000,
```
Comment (jbrooksuk) [inline app/Foundation/Providers/ModuleServiceProvider.php]: Extra space.

Comment (ConnorVG) [inline app/Foundation/Providers/ModuleServiceProvider.php]: This is a StyleCI bug? Unless there is something I'm not seeing.

File: app/helpers.php [hunk 3]
```diff
@@ -131,7 +131,7 @@ function array_numeric_sort(array $array = [], $key = 'order')
             $a = array_get($a, $key, PHP_INT_MAX);
             $b = array_get($b, $key, PHP_INT_MAX);
 
-            $default = PHP_MAJOR_VERSION < 7 ? 1 : 0;
+            $default = PHP_MAJOR_VERSION < 7 && !defined('HHVM_VERSION') ? 1 : 0;
```
Comment (GrahamCampbell) [inline app/helpers.php]: We're dropping HHVM support soon. Might just do it in 2.4.

Comment (ConnorVG) [inline app/helpers.php]: fuck hhvm

Comment (jbrooksuk) [inline app/helpers.php]: 🎤 ⬇️

File: app/helpers.php [hunk 4]
```diff
@@ -115,3 +115,27 @@ function color_contrast($hexcolor)
...
+     *
+     * @return array
+     */
+    function array_numeric_sort(array $array = [], $key = 'order')
+    {
+        uasort($array, function ($a, $b) use ($key) {
+            $a = array_get($a, $key, PHP_INT_MAX);
+            $b = array_get($b, $key, PHP_INT_MAX);
+
+            $default = PHP_MAJOR_VERSION < 7 && !defined('HHVM_VERSION') ? 1 : 0;
```
Comment (jbrooksuk) [inline app/helpers.php]: We can drop the check for `HHVM_VERSION`.

Comment (ConnorVG) [inline app/helpers.php]: Will do.

**Comment by jbrooksuk** [comment]
Can you rebase please?

**Comment by ConnorVG** [comment]
That'll have to do unless you want it better.

#WorkStress

**Comment by jbrooksuk** [comment]
💥

**Pull request merged**